metrics/chronos_httpserver.nim

```diff
@@ -305,49 +305,54 @@ proc new*(t: typedesc[MetricsHttpServerRef], address: string,
           initTAddress(address, port)
         except TransportAddressError as exc:
           return err("Invalid server address")
+    var
       request =
         block:
           let res = createAsyncPipe()
           if (res.read == asyncInvalidPipe) or (res.write == asyncInvalidPipe):
             return err("Unable to create communication request pipe")
           res
+      cleanupRequest = true
+    defer:
```

Comment:
I'm not convinced this defer is a win when reading the code - ie the booleands do quite a good job of obfuscating the control flow, which also can be seen in the line count

Reply:
I strongly disagree. With this structure in place it's trivial to modify the control flow in the function in safe manner. Without it, one has to be extremely careful to keep all the exit paths correct.

Reply:
You should appreciate this classic article and the arguments that it puts forward:
http://ptgmedia.pearsoncmg.com/imprint_downloads/informit/aw/meyerscddemo/DEMO/MAGAZINE/CA_FRAME.HTM

Resource management without RAII-like patterns in just not viable.